database/functions.js

```diff
@@ -217,58 +217,50 @@ const getDrugData = async function(health_zone_id, numPastDays) {
-            // every report instance has a health_area property. We implemented to the rendundancy to reducy queries to the database
-            const reports = await Report.find({ health_area: healthArea, is_validated: true, 'date': {'$gte': new Date(earliestDate)} });
-            if (reports.length != 0) {
-                // these arrays hold the data about the drugs in the Drug Management section of a report
-                const drugsName = ["ivermectin", "albendazole", "praziquantel"];
-                const drugsUsed = [];
-                const drugsReceived = [];
-                
-                for (let i = 0; i < drugsName.length; i++) {
-                    drugsUsed.push(0);
-                    drugsReceived.push(0);
-                }
+        const reports = await Report.find({ health_zone: health_zone_id, is_validated: true, 'date': {'$gte': new Date(earliestDate)} });
+        if (reports.length != 0) {
+            // these arrays hold the data about the drugs in the Drug Management section of a report
+            const drugsName = ["ivermectin", "albendazole", "praziquantel"];
+            const drugsUsed = [];
+            const drugsReceived = [];
+            
+            for (let i = 0; i < drugsName.length; i++) {
+                drugsUsed.push(0);
+                drugsReceived.push(0);
+            }
 
-                // get the drug data needed from each valid report
-                for (r in reports) {
-                    report = reports[r];
-                    for (let i = 0; i < drugsName.length; i++) {
-                        let drug_management = drugsName[i] + "_management";
-                        drugsUsed[i] += report[drug_management]["quantityUsed"];
-                        drugsReceived[i] += report[drug_management]["quantityReceived"];
-                    }
+            // get the drug data needed from each valid report
+            for (r in reports) {
+                report = reports[r];
+                for (let i = 0; i < drugsName.length; i++) {
+                    let drug_management = drugsName[i] + "_management";
+                    drugsUsed[i] += report[drug_management]["quantityUsed"];
+                    drugsReceived[i] += report[drug_management]["quantityReceived"];
                 }
+            }
 
 
-                // calculate the drug percentages/proportions used for each drug for a health area, and update drugData object
-                drugData[healthArea.name] = {};
-                for (let i = 0; i < drugsName.length; i++) {
-                    let drugPercentage = Math.round(drugsUsed[i] / (drugsReceived[i]||1) * 100);
-                    console.log(drugPercentage);
-                    drugData[healthArea.name][drugsName[i]] = drugPercentage;
-                }
+            // calculate the drug percentages/proportions used for each drug for a health area, and update drugData object
+            drugData[healthArea.name] = {};
```

Comment:
I was trying to test this, but it's raising an error about this variable: healthArea is not defined. I don't know if you tested this, but I can't see where this is declared. When you're done, test this endpoint with the healthzone id, 618b21eb8453970bd916764c, and see if you get anything back. If it works, you should get something back.